src/base/bittorrent/session.cpp

```diff
@@ -2118,9 +2118,10 @@ void Session::enableIPFilter(const QString &filterPath, bool force)
         connect(m_filterParser.data(), SIGNAL(IPFilterParsed(int)), SLOT(handleIPFilterParsed(int)));
         connect(m_filterParser.data(), SIGNAL(IPFilterError()), SLOT(handleIPFilterError()));
     }
-    if (m_filterPath.isEmpty() || m_filterPath != Utils::Fs::fromNativePath(filterPath) || force) {
-        m_filterPath = Utils::Fs::fromNativePath(filterPath);
-        m_filterParser->processFilterFile(Utils::Fs::fromNativePath(filterPath));
+    const QString& newFilterPath=Utils::Fs::fromNativePath(filterPath);
+    if (m_filterPath.isEmpty() || m_filterPath != newFilterPath || force) {
```

Comment:
Parenthesis around `(m_filterPath != newFilterPath)` for clarity. There is no rule for this but i was corrected once if i remember correctly.